target/linux/ramips/patches-4.4/0902-fix-reboot-hang-with-32m-spi-flash.patch

```diff
@@ -0,0 +1,69 @@
+diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
+index b298466..932f307 100644
+--- a/drivers/mtd/devices/m25p80.c
++++ b/drivers/mtd/devices/m25p80.c
+@@ -21,6 +21,7 @@
+ #include <linux/module.h>
+ #include <linux/device.h>
+ #include <linux/interrupt.h>
++#include <linux/reboot.h>
+ #include <linux/mutex.h>
+ #include <linux/math64.h>
+ #include <linux/slab.h>
+@@ -61,6 +62,9 @@
+ /* Used for Spansion flashes only. */
+ #define	OPCODE_BRWR		0x17	/* Bank register write */
+ 
++#define OPCODE_RESET_ENABLE	0x66
++#define OPCODE_RESET		0x99
++
+ /* Status Register bits. */
+ #define	SR_WIP			1	/* Write in progress */
+ #define	SR_WEL			2	/* Write enable latch */
+@@ -907,6 +911,21 @@ static const struct spi_device_id *jedec_probe(struct spi_device *spi)
+ 	return ERR_PTR(-ENODEV);
+ }
+ 
++static int m25p80_reboot(struct notifier_block *nb, unsigned long val,
++			       void *v)
++{
++	struct mtd_info *mtd;
++
++	mtd = container_of(nb, struct mtd_info, reboot_notifier);
```

Comment:
this patch look technically correct, however could add a

if (this is a chip that needs the quirk) {

}

line so that the registers are only written if we really need to.